ads/google/utils.js

```diff
@@ -34,22 +34,31 @@ export const ADSENSE_RSPV_WHITELISTED_HEIGHT = 320;
  * @param {boolean} multiSizeValidation A flag that if set to true will enforce
  *   the rule that ensures multi-size dimensions are no less than 2/3rds of
  *   their primary dimension's counterpart.
- * @param {boolean=} isFluid Indicates whether this ad slot is Fluid-enabled.
+ * @param {boolean=} isFluidPrimary Indicates whether the ad slot's primary
+ *   size is fluid.
  * @return {?Array<!Array<number>>} An array of dimensions.
  */
 export function getMultiSizeDimensions(
   multiSizeDataStr,
   primaryWidth,
   primaryHeight,
   multiSizeValidation,
-  isFluid = false) {
+  isFluidPrimary = false) {
 
   const dimensions = [];
   const arrayOfSizeStrs = multiSizeDataStr.split(',');
 
   for (let i = 0; i < arrayOfSizeStrs.length; i++) {
 
     const sizeStr = arrayOfSizeStrs[i];
+    if (sizeStr.toLowerCase() == 'fluid') {
+      if (!isFluidPrimary) {
+        // If the primary size is fluid, then the dummy 320x50 size will
+        // automatically be included.
+        dimensions.push([320, 50]);
```

Comment:
That the dummy size to indicate fluid is 320x50 should probably be in a (pair of?) constants, instead of scattered around the code.